From: Matthias Klose <doko@ubuntu.com>
Subject: Fix support for Python 3.5, adding PyAsyncMethods

  Python 3.5 adds new PyAsyncMethods to the PyTypeObject, this patch adjusts
  the code that SWIG generates when using the -builtin flag.

Origin: http://launchpadlibrarian.net/222739670/swig_3.0.7-0ubuntu1_3.0.7-0ubuntu2.diff.gz
Bug: https://github.com/swig/swig/issues/539
Bug-Debian: http://bugs.debian.org/802906
Applied-Upstream: 3.0.8, https://github.com/swig/swig/commit/ef001de5240c1e05494e23b933b687f3f266045c

Index: b/Source/Modules/python.cxx
===================================================================
--- a/Source/Modules/python.cxx
+++ b/Source/Modules/python.cxx
@@ -4053,6 +4053,15 @@ public:
     Printv(f, "#endif\n", NIL);
     Printf(f, "  },\n");
 
+    // PyAsyncMethods as_async
+    Printv(f, "#if PY_VERSION_HEX >= 0x03050000\n", NIL);
+    Printf(f, "  {\n");
+    printSlot(f, getSlot(n, "feature:python:am_await"), "am_await", "unaryfunc");
+    printSlot(f, getSlot(n, "feature:python:am_aiter"), "am_aiter", "unaryfunc");
+    printSlot(f, getSlot(n, "feature:python:am_anext"), "am_anext", "unaryfunc");
+    Printf(f, "  },\n");
+    Printv(f, "#endif\n", NIL);
+
     // PyNumberMethods as_number
     Printf(f, "  {\n");
     printSlot(f, getSlot(n, "feature:python:nb_add"), "nb_add", "binaryfunc");
